app/code/Magento/Catalog/Controller/Adminhtml/Product/Initialization/Helper.php

```diff
@@ -90,14 +95,16 @@ public function __construct(
         StockDataFilter $stockFilter,
         \Magento\Catalog\Model\Product\Initialization\Helper\ProductLinks $productLinks,
         \Magento\Backend\Helper\Js $jsHelper,
-        \Magento\Framework\Stdlib\DateTime\Filter\Date $dateFilter
+        \Magento\Framework\Stdlib\DateTime\Filter\Date $dateFilter,
+        \Magento\Catalog\Model\Product\LinkTypeProvider $linkTypeProvider
```

Comment:
It seems to be a bit counter intuitive to use ObjectManager here when "the good practices" tell you to avoid touching it, specially for a class that is not part of the `@api`.